### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by mmorhun**
This PR updates:
- images used by the operator
- operator dependencies

### CODE REVIEW

File: controllers/component_build_controller.go [hunk 1]
```diff
@@ -217,6 +217,9 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, nil
 	}
 
+	log = log.WithValues("ComponentGitSource", component.Spec.Source.GitSource)
+	ctx = ctrllog.IntoContext(ctx, log)
+
```
Comment (tkdchen) [inline controllers/component_build_controller.go:222]: Is this necessary for the update?

File: Dockerfile [hunk 2]
```diff
@@ -18,9 +20,16 @@ COPY pkg/ pkg/
...
+# https://catalog.redhat.com/software/containers/ubi9/ubi-minimal/615bd9b4075b022acc111bf5
+FROM registry.access.redhat.com/ubi9/ubi-minimal:9.3
 COPY --from=builder /opt/app-root/src/manager /
 USER 65532:65532
 
+LABEL description="RHTAP Build Service operator"
+LABEL io.k8s.description="RHTAP Build Service operator"
+LABEL io.k8s.display-name="build-service-operator"
+LABEL io.openshift.tags="rhtap"
+LABEL summary="RHTAP Build Service"
```
Comment (chmeliik) [inline Dockerfile:33]: Could the labels be in a separate commit?

File: config/default/manager_auth_proxy_patch.yaml [hunk 3]
```diff
@@ -15,8 +15,9 @@ spec:
         - "--metrics-bind-address=127.0.0.1:8080"
         - "--leader-elect"
       - name: kube-rbac-proxy
-        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.13.0
+        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.15.0
         args:
+        - "--http2-disable"
```
Comment (chmeliik) [inline config/default/manager_auth_proxy_patch.yaml:20]: I guess it's still not safe to explain why we're adding `--http2-disable`? :disappointed:

File: main.go [hunk 4]
```diff
@@ -199,22 +202,20 @@ func getCacheExcludedObjectsTypes() []client.Object {
 	}
 }
 
-func getCacheFuncOptions() (*cache.Options, error) {
+func getCacheOptions() cache.Options {
 	componentPipelineRunRequirement, err := labels.NewRequirement(controllers.ComponentNameLabelName, selection.Exists, []string{})
 	if err != nil {
-		return nil, err
+		panic(err)
```
Comment (chmeliik) [inline main.go:208]: Why switch from returning an error to `panic`?

Comment (mmorhun) [reply main.go:208]: Because the operation above is safe. It can error if we configure it wrongly, but that should be fixed in the same commit and shouldn't bother the caller.

File: controllers/component_build_controller.go (hunk 1 above)
Comment (mmorhun) [reply controllers/component_build_controller.go:222]: Not really. It's just a minor improvement for logging.

File: main.go (hunk 4 above)
Comment (chmeliik) [reply main.go:208]: Ack :+1: 

Could you add a comment about that or explain it in the commit message?

File: controllers/component_build_controller.go (hunk 1 above)
Comment (mkosiarc) [reply controllers/component_build_controller.go:222]: I would also include it in separate commit in that case

Comment (mmorhun) [reply controllers/component_build_controller.go:222]: ok

File: Dockerfile (hunk 2 above)
Comment (mmorhun) [reply Dockerfile:33]: ok

File: main.go (hunk 4 above)
Comment (mmorhun) [reply main.go]: ok

**Review by chmeliik — APPROVED** [review]
LGTM :+1: 

CI needs a rerun

File: main.go [hunk 5]
```diff
@@ -109,21 +110,23 @@ func main() {
...
-	if err != nil {
-		setupLog.Error(err, "unable to create cache options")
-		os.Exit(1)
+	clientOpts := client.Options{
+		Cache: &client.CacheOptions{
+			DisableFor: getCacheExcludedObjectsTypes(),
+		},
+	}
+	metricsOpts := server.Options{
+		BindAddress: metricsAddr,
```
Comment (mkosiarc) [inline main.go:119]: How is this code change related to the updated operator dependencies?

Comment (mmorhun) [reply main.go:119]: Breaking changes in the configuration structures.

**Review by mkosiarc — APPROVED** [review]
